internal/global/state_test.go

```diff
@@ -12,36 +12,91 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package global_test
+package global
 
 import (
 	"testing"
 
-	"go.opentelemetry.io/otel/internal/global"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/trace"
 )
 
-func TestResetsOfGlobalsPanic(t *testing.T) {
-	global.ResetForTest()
-	tests := map[string]func(){
-		"SetTextMapPropagator": func() {
-			global.SetTextMapPropagator(global.TextMapPropagator())
-		},
-		"SetTracerProvider": func() {
-			global.SetTracerProvider(global.TracerProvider())
-		},
-	}
-
-	for name, test := range tests {
-		shouldPanic(t, name, test)
-	}
+func TestSetTracerProvider(t *testing.T) {
+	t.Cleanup(ResetForTest)
```

Comment:
I think that ResetForTest could call t.Cleanup instead of requirying the caller to do it

Reply:
I'll be doing that in another PR, as it's unrelated to those changes.

Reply:
See #2754